### PR TITLE
Fix Issue #1136 (issue of mqtt-client example + rpl-lite)

### PR DESCRIPTION
**Pull request by yatch**
See Issue #1136 for details. 

@g-oikonomou Can you confirm this fix works on your side? I did a test using the `.csc` files attached to Issue #1136 .

### CODE REVIEW

**Comment by g-oikonomou** [comment]
The fix works yes.

File: examples/mqtt-client/mqtt-client.c [hunk 1]
```diff
@@ -591,12 +601,12 @@ connect_to_broker(void)
 static void
 ping_parent(void)
 {
-  if(uip_ds6_get_global(ADDR_PREFERRED) == NULL) {
-    return;
+  if(have_connectivity()) {
+    uip_icmp6_send(uip_ds6_defrt_choose(), ICMP6_ECHO_REQUEST, 0,
+                   ECHO_REQ_PAYLOAD_LEN);
+  } else {
+    /* we don't have connectivity; cannot send a ping */
```
Comment (g-oikonomou) [inline examples/mqtt-client/mqtt-client.c]: Do we need the else block here? (Unless we add some logging)

Comment (yatch) [reply examples/mqtt-client/mqtt-client.c]: Let me add `LOG_ERR` here :-)

Comment (g-oikonomou) [reply examples/mqtt-client/mqtt-client.c]: I wouldn't call this quite an error. More of a "not ready yet". I'd either warn or dbg.

Comment (yatch) [reply examples/mqtt-client/mqtt-client.c]: oh, alright!

**Comment by yatch** [comment]
@g-oikonomou Thank you for the review. I put `LOG_WARN` and performed force-push.

**Comment by g-oikonomou** [comment]
OK 👍 will merge when back near the puter, unless someone else gets there first. Thanks!

**Pull request merged**